app/src/main/java/org/mozilla/fenix/components/Components.kt

```diff
@@ -143,8 +144,12 @@ class Components(private val context: Context) {
         AddonManager(core.store, core.engine, addonCollectionProvider, addonUpdater)
     }
 
+    val wallpaperStorage by lazyMonitored {
+        WallpapersAssetsStorage(context)
+    }
```

Comment:
I think this could be inlined in the `wallpaperManager` creation, as I don't think it will be used by other objects.

Reply:
This could make testing difficult as we won't have control over the `wallpaperStorage` passing it as dependency we won't have to use the real object for testing.

Reply:
Sorry, I meant inlined in the call to the constructor itself. That way it we still have the ability to inject mocked or faked dependencies during test:

```
val wallpaperManager by lazyMonitored {
        WallpaperManager(settings, WallpapersAssetsStorage(context))
    }
}
```

Reply:
Thanks for the clarification 👍🏽

Reply:
Updated!